File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 5/20/2026, 11:37:34 PM


class Solution {
public:
  int longestConsecutive(vector<int> &nums) {
    if(nums.empty()) return 0;
    std::sort(nums.begin(), nums.end());
    int maxlen{};
    int currentLen{1};
    for(int i = 1; i < nums.size(); ++i) {
        if(nums[i - 1] == nums[i]) {
            continue;
        }
        if(nums[i] == nums[i - 1] + 1) {
            currentLen++;
        } else {
            maxlen = std::max(maxlen, currentLen);
            currentLen = 1;
        }
    }
    maxlen = std::max(maxlen, currentLen);
    return maxlen;
  }
};

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.js
// Last updated: 5/20/2026, 11:37:31 PM
/*
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
    if(nums.length === 0) return 0;
    let longestStreak = 1;
    nums.sort((a, b) => a - b);
    let current = 1;
    for (let i = 0; i < nums.length - 1; ++i) {
        if (nums[i] === nums[i + 1]) continue;
        if (nums[i + 1] - nums[i] === 1) {
            ++current;
        } else {
            longestStreak = Math.max(longestStreak, current);
            current = 1;
        }
    }

    longestStreak = Math.max(longestStreak, current);
    return longestStreak;
};


